### ISSUE TITLE
The notification of "server certificate expired  send too many

### ISSUE THREAD

**Issue by raydoom** [louislam/uptime-kuma#4520]
### ⚠️ Please verify that this question has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

The notification of "server certificate expired " send too many ,will send a message about 2-6min 
![E66C74F3-8C4F-4C22-9DE5-CB7198C2DD7E_1_201_a](https://github.com/louislam/uptime-kuma/assets/26057967/7e8d9fc7-0eae-46fa-93cd-248a44e5f49d)

### 🐻 Uptime-Kuma Version

1.23.11

### 💻 Operating System and Arch

Rocky linux 8

### 🌐 Browser

google chrome 121.0.6167.184

### 🖥️ Deployment Environment

- Runtime: docker 24.0.6
- Database: sqlite
- Filesystem used to store the database on: NFSv3 on a SSD from synology nas
- number of monitors: 12

**Comment by raydoom** [comment]
I have two monitors,one is www.example1.com ,and another is www.example2.com which redirect to www.example1.com by nginx with code 302
TLS Certificate Expiry:
<img width="621" alt="image" src="https://github.com/louislam/uptime-kuma/assets/26057967/c72bca24-c734-4979-8ea4-b2e9f7086187">

**Comment by CommanderStorm** [comment]
Let me rephrase:
The list of notifications you are getting, to which of these domains are they pointing to?
Are they pointing to different or the same domain?

**Comment by raydoom** [comment]
the message is:
```
UptimeKuma Message
[www.example1.com][https://www.example1.com] server certificate *.example2.com will be expired in 17 days
```
the cert expire info:
*.example1.com: remian 312 days
*.example2.com: remian 17 days

**Comment by Suplanus** [comment]
Same here.
The expire date is flapping (lets encrypt).
"Somewhere" the old certificate is in kuma. If I look via Browser or into my reverse proxy (HaProxy), the certificates are new.

Kuma are seeing the new certificate but sometimes shows up the old with the expire date.
Example: https://md3.page

The notification is not send only once. Its sends like descriped every 2 minutes.

**Comment by CommanderStorm** [comment]
@Suplanus 
Yes, but that is because your service keeps flapping between thos expiry dates.
I don't see this as a bug or as something that we can improve on our side..

```
curl -s -v -X GET https://md3.page
```

| July | May |
|--------|--------|
| ![image](https://github.com/louislam/uptime-kuma/assets/26258709/657ca487-ef21-4d04-b76c-b413b32eece3) | ![image](https://github.com/louislam/uptime-kuma/assets/26258709/506f73b9-1c5f-400e-8a3f-ff210809a252) |

I don't think this misconfiguration is common enough to warrant extra handling.
I am going to close this as not planned.

If you have a good rationale why this should be supported or how to support this, we can reopnen.

**Comment by Suplanus** [comment]
@CommanderStorm Thanks for looking into it. I will fix it on my side.
But: I think the notification should only be sent once per setting of days.

**Comment by CommanderStorm** [comment]
Checking the cert is not expensive. Doing so on every request is fine.
I don't see an upside from introducing more cases to test, even downsides (would you have found the bug before May otherwise?)

**Comment by Suplanus** [comment]
I don't changed the setup (OpnSense -> HaProxy -> ACME)... And yes, the problem is since may.
I looked into OpnSense and there is only one certificate of each domain...

So I am still digging :)

**Comment by Suplanus** [comment]
For all other with the same problem:  
TLDR: It's not a kuma problem, but the great kuma shows it for us :)

Its a Bug in HAProxys Lets Encrypt implementation: https://forum.opnsense.org/index.php?topic=38435.0
A Update is available.
